docs/docs/building-modules/02-messages-and-queries.md

```diff
@@ -38,6 +38,40 @@ Each `Msg` service method must have exactly one argument, which must implement t
 
 `sdk.Msg` interface is a simplified version of the Amino `LegacyMsg` interface described [below](#legacy-amino-msgs) with the `GetSigners()` method. For backwards compatibility with [Amino `LegacyMsg`s](#legacy-amino-msgs), existing `LegacyMsg` types should be used as the request parameter for `service` RPC definitions. Newer `sdk.Msg` types, which only support `service` definitions, should use canonical `Msg...` name.
 
+`sdk.Msg` is a alias of `proto.Message`. To attach a `ValidateBasic()` method to a message then you must add methods to the type adhereing to the `HasValidateBasic`.
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/9c1e8b247cd47b5d3decda6e86fbc3bc996ee5d7/types/tx_msg.go#L84-L89
+```
+
+In 0.50+ signers from the `GetSigners()` call is automated via a protobuf annotation. 
```

Comment:
It would be great to have this as well, to be able to link it here: https://github.com/cosmos/cosmos-sdk/pull/15548. All mentions of it are undocumented.
![image](https://github.com/cosmos/cosmos-sdk/assets/29894366/07133349-f208-4be3-b608-79c1e6a117f0)

Reply:
ill leave a todo then reopen that pr